http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnection.java

```diff
@@ -37,7 +37,13 @@
    * If sending messages before a connection is opened, they will be queued. When the connection is
    * opened, the queue is flushed and messages will be sent in order.
    */
-  private final Queue<String> queuedMessages = new ConcurrentLinkedQueue<>();
+  private final Queue<String> queuedMessages = new LinkedList<>();
```

Comment:
Went with linked list, constant time append and not having to pre-allocate space are positives.

Reply:
Error-prone will likely fail.
Even though it's faster in theory, modern CPUs are just better at allocating blocks instead of random memory

Reply:
Sure enough:

> /home/travis/build/triplea-game/triplea/http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnection.java:40: error: [JdkObsolete] It is very rare for LinkedList to out-perform ArrayList or ArrayDeque. Avoid it unless you're willing to invest a lot of time into benchmarking. Caveat: LinkedList supports null elements, but ArrayDeque does not.